guides/store-retrieve-messages.md

```diff
@@ -0,0 +1,162 @@
+# Retrieve Messages Using Waku Store
+
+When running a web dApp or a mobile phone application,
+internet can be unreliable and disconnect.
+
+[Waku Relay](https://rfc.vac.dev/spec/18/) is a gossip protocol.
+As a user, it means that your peers forward you messages they just received.
+If you cannot be reached by your peers, then messages are not relayed;
+relay peers do **not** save messages for later.
+However, [store](https://rfc.vac.dev/spec/13/) peers do save messages they relay,
+allowing you to retrieve them at any time.
+
+In this guide, we'll review how you can use Waku Store to retrieve messages.
+
+Before starting, you need to choose a _Content Topic_ for your dApp.
+Check out the [how to choose a content topic guide](choose-content-topic.md) to learn more about content topics.
+
+For this guide, we are using a single content topic: `/store-guide/1/news/proto`.
+
+# Installation
+
+You can install [js-waku](https://npmjs.com/package/js-waku) using your favorite package manager:
+
+```shell
+npm install js-waku
+```
+
+# Create Waku Instance
+
+In order to interact with the Waku network, you first need a Waku instance:
+
+```js
+import { Waku } from 'js-waku';
+
+const wakuNode = await Waku.create();
+```
+
+# Connect to Other Peers
+
+The Waku instance needs to connect to other peers to communicate with the network.
+You are free to choose other methods to bootstrap and DappConnect will ship with new bootstrap mechanisms in the future.
+
+For now, the easiest way is to connect to Status' Waku fleet:
+
+```js
+import { getStatusFleetNodes } from 'js-waku';
+const nodes = await getStatusFleetNodes();
```

Comment:
Think this was brought up before, we don't want a specific Status function here. Instead, it should be

`getBootstrapNodes` or similar, which points to a static file, where one comment is:

```
// Status fleet nodes (see <URL>)
node1
ndoe2
node3

// Community nodes
...
```

Status has no special role here and we should make sure we stay consistent with this vision

Reply:
So instead of retrieving nodes from fleets.status.im, you suggest to save the nodes multiaddresses in a static file.

Are we confident that these multiaddresses will not change?

Reply:
Right hm, so this should be addressed with DNS discovery, where `fleets.status.im`. Can we do something similar where we mock this for now? And a comment that this is temporary. The point is to not enshrine things like `getStatusNodes` in API or docs as it gives Status a weird status.

I.e. `getBootstrapNodes`

```
//TODO: Query multiple fleets with DNS discovery, for now just using fleets.status.im
var entries = ["fleets.status.im"]
<logic>
```

Reply:
> Right hm, so this should be addressed with DNS discovery, where `fleets.status.im`. Can we do something similar where we mock this for now? And a comment that this is temporary. The point is to not enshrine things like `getStatusNodes` in API or docs as it gives Status a weird status.
> 
> I.e. `getBootstrapNodes`
> 
> ```
> //TODO: Query multiple fleets with DNS discovery, for now just using fleets.status.im
> var entries = ["fleets.status.im"]
> <logic>
> ```

Looks good, thanks for the alternative solution. Tracking here: https://github.com/status-im/js-waku/issues/261 and will tackle first.